src/controller/python/chip/yaml/parser.py

```diff
@@ -170,18 +337,28 @@ def __init__(self, item: dict, cluster: str, data_model_lookup: DataModelLookup)
             raise UnexpectedParsingError(
                 f'ReadAttribute doesnt have valid attribute_type. {self.label}')
 
+        if 'optional' in item:
+            self._possibly_unsupported = True
+
         self._expected_raw_response = item.get('response')
         if (self._expected_raw_response is None):
```

Comment:
I know this isn't part of this change, but does this cause an error if you have a ReadAttribute with no response: tag? That's syntactically valid - if you have 
  - label: "whatever"
      command: "readAttribute"
      attribute: "whatever"

it means read the "whatever" attribute and ensure we get success response. I think this appears for some of the client side tests.

Reply:
A lot of the `UnexpectedParsingError` are just area that I need to fine tune. I added a todo comment here